sips/committee_consensus.md

```diff
@@ -0,0 +1,480 @@
+		binary.LittleEndian.PutUint32(bytes[i*4:], uint32(v))
+	}
+	// Hash
+	return CommitteeID(sha256.Sum256(bytes))
+}
+```
+
+In order to route consensus messages to the correct consensus runner, the `CommitteeID` field will be included in the `MessageID` replacing `ValidatorPublicKey`.
+
+### Role
+
+This is used to route the message to the correct runner. Since `CommitteeRunner` has several beacone roles we will create a new `RunnerRole` in the MessageID.
+
+```go
+type RunnerRole int32
+
+const (
+	RoleCommittee RunnerRole = iota
+	RoleAggregator
+	RoleProposer
+	RoleSyncCommitteeContribution
+
+	RoleValidatorRegistration
+	RoleVoluntaryExit
+
+	RoleUnknown = -1
+)
+```
+
+### MessageID
+
+`ValidatorPublicKey` and `CommitteeID` will be used interchangeably in `MessageID`. `Role` will change location because it is used to determine ID type.  `CommitteeID` is 32 bytes long so it will be encoded with a `0x00` 16 bytes prefix to make it the same length as `ValidatorPublicKey`.
+
+```go
+const (
+	domainSize     = 4
+	domainStartPos = 0
+	roleTypeSize   = 4
+	// CHANGE IN Positions
+	roleTypeStartPos = domainStartPos + domainSize
+	senderIDSize     = 48
+	senderIDStartPos = roleTypeStartPos + roleTypeSize
+)
+
+// MessageID is used to identify and route messages to the right validator and Runner
+type MessageID [56]byte
+
+func (msg MessageID) GetDomain() DomainType {
+	return msg[domainStartPos : domainStartPos+domainSize]
+}
+
+func (msg MessageID) GetRoleType() RunnerRole {
+	roleByts := msg[roleTypeStartPos : roleTypeStartPos+roleTypeSize]
+	return BeaconRole(binary.LittleEndian.Uint32(roleByts))
+}
+
+func (msg MessageID) GetSenderID() []byte {
+	return msg[senderIDStartPos : senderIDStartPos+senderIDSize]
+}
+
+func parseMessageID(messageID MessageID) (DomainType, BeaconRole, []byte) {
+	domain := messageID.GetDomain()
+	role := messageID.GetRoleType()
+	if role == BNRoleSyncCommitteeContribution {
+		return domain, BNRoleAttesterOrSyncCommittee, messageID.GetSenderID()[16:]
+	}
+	return domain, role, messageID.GetSenderID()
+}
+```
+
+
+### Consensus Data
+
+We note that the data needed for a consensus execution is the same for all validators. Thus, we can create a `BeaconVote` object that will hold the data for all validators. The data needed for the `SyncCommittee` role is a subset of the data needed for the `Attestation` role. Thus we can always query the beacon node for attestation data and pass this data to relevant runners.
+
+`ConsensusData` currently holds the `duty` field to make sure that all committee members agree on committee information. However, if there is a difference between committee members there is no guarantee that a run will be triggered on the first place. This is because different views may cause [different shuffles](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#compute_committee). Therefore we can rely on local view of `duty`, and keep the field empty. 
```

Comment:
need to re-write this, explain consensus data is the old struct and is left because of the old runners